Objects/exceptions.c

```diff
@@ -2897,8 +2897,24 @@ _check_for_legacy_statements(PySyntaxErrorObject *self, Py_ssize_t start)
     }
     if (PyUnicode_Tailmatch(self->text, print_prefix,
                             start, text_len, -1)) {
-        Py_XSETREF(self->msg,
-                  PyUnicode_FromString("Missing parentheses in call to 'print'"));
+        char *error_msg_start = "Missing parentheses in call to 'print'. Did you mean 'print(";
+        // PRINT_OFFSET is to remove print word from the data.
+        const int PRINT_OFFSET = 6;
+        char *msg = data + PRINT_OFFSET;
+        char *error_msg_end = ")'?";
+
+        char *error_msg = malloc(strlen(error_msg_start) + \
```

Comment:
1. This should use `PyMem_Malloc` and `PyMem_Free` rather than the direct C APIs. Also, it should actually free the allocated working memory :)
2. We may want to include a short block comment explaining that `data` is UTF-8 encoded, so we mostly treat it as an indivisible opaque blob, except that we know it starts with `print ` and *might* end with `,`. This is also why we work directly with `char *` and convert to a Python Unicode object at the end, rather than pre-allocating the object via `PyUnicode_New` and populating it incrementally.